api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -1167,6 +1167,21 @@ message HttpFilter {
   // filter but otherwise accept the config.
   // Otherwise, clients that do not support this filter must reject the config.
   bool is_optional = 6;
+
+  // [#not-implemented-hide:]
+  // If true, the filter is disabled by default and must be explicitly enabled by explicitly setting per
+  // filter configuration in the route configuration.
+  //
+  // A disabled filter will be treated as enabled in following cases:
+  //
+  // 1. Valid per filter configuration is configured in the ``typed_per_filter_config`` of route config.
+  // 2. :ref:`FilterConfig <envoy_v3_api_msg_config.route.v3.FilterConfig>` is configured in the
+  //    ``typed_per_filter_config`` of route config and the
+  //    :ref:`disabled of FilterConfig <envoy_v3_api_field_config.route.v3.FilterConfig.disabled>`
+  //    is set to false.
+  //
+  // Terminal filters (e.g. ``envoy.filters.http.router``) should never be marked as disabled by default.
```

Comment:
Can we just make this a config error? So I would just say terminal filters cannot be marked disabled by default.

Reply:
Of course. :)